invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,20 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"] == "L":
+                data.pop("uri")
+
+            # [TODO] Remote and Fetch
```

Comment:
Why this TODO? Is it because you are not sure what should be the wanted behaviour here?
If that's the case, it looks like that the answers are in the issue ;)

Reply:
I put it as a TODO "discussion"
But yes for both of the cases (Fetch and Remote) there should be no need to modify the URI
I'll remove them